### PR TITLE
chore(storage): add more log info when panic or assert for compactor component

### PR DESCRIPTION
**Pull request by Li0k**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

We often encounter panic but are troubled by being unable to locate the table or compact task. This PR attempts to add more log info for assert and panic.

## Checklist

- [ ] I have written necessary rustdoc comments
- [ ] I have added necessary unit tests and integration tests
- [ ] I have added test labels as necessary. See [details](https://github.com/risingwavelabs/risingwave/blob/main/docs/developer-guide.md#ci-labels-guide).
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

- [ ] My PR contains critical fixes that are necessary to be merged into the latest release. (Please check out the [details](https://github.com/risingwavelabs/risingwave/blob/main/CONTRIBUTING.md))

## Documentation

- [ ] My PR needs documentation updates. (Please use the **Release note** section below to summarize the impact on users)

## Release note

If this PR includes changes that directly affect users or other significant modifications relevant to the community, kindly draft a release note to provide a concise summary of these changes. Please prioritize highlighting the impact these changes will have on users.

</details>

### CODE REVIEW

**Comment by Li0k** [comment]
If you have better suggestions, or need to print more information, please let me know

File: src/storage/src/hummock/sstable/builder.rs [hunk 1]
```diff
@@ -256,8 +256,12 @@ impl<W: SstableWriter, F: FilterBuilder> SstableBuilder<W, F> {
                 let mut iter = BlockIterator::new(BlockHolder::from_owned_block(Box::new(block)));
                 iter.seek_to_first();
                 while iter.is_valid() {
-                    let value = HummockValue::from_slice(iter.value())
-                        .expect("decode failed for fast compact");
+                    let value = HummockValue::from_slice(iter.value()).unwrap_or_else(|_| {
+                        panic!(
+                            "decode failed for fast compact last_table_id {:?}",
+                            self.last_table_id
```
Comment (hzxa21) [inline src/storage/src/hummock/sstable/builder.rs:262]: Is it possible to include the sst id and block id as well?

File: src/storage/src/hummock/sstable/builder.rs [hunk 2]
```diff
@@ -323,7 +334,11 @@ impl<W: SstableWriter, F: FilterBuilder> SstableBuilder<W, F> {
         let table_id = full_key.user_key.table_id.table_id();
         let is_new_table = self.last_table_id.is_none() || self.last_table_id.unwrap() != table_id;
         if is_new_table {
-            assert!(could_switch_block);
+            assert!(
+                could_switch_block,
```
Comment (hzxa21) [inline src/storage/src/hummock/sstable/builder.rs:338]: Is it possible to include the sst id and block id as well?

File: src/storage/src/hummock/sstable/builder.rs [hunk 3]
```diff
@@ -342,7 +357,13 @@ impl<W: SstableWriter, F: FilterBuilder> SstableBuilder<W, F> {
         // Rotate block builder if the previous one has been built.
         if self.block_builder.is_empty() {
             self.block_metas.push(BlockMeta {
-                offset: utils::checked_into_u32(self.writer.data_len()),
+                offset: utils::checked_into_u32(self.writer.data_len()).unwrap_or_else(|_| {
+                    panic!(
+                        "WARN overflow can't convert writer_data_len {} into u32 tables {:?}",
+                        self.writer.data_len(),
```
Comment (hzxa21) [inline src/storage/src/hummock/sstable/builder.rs:363]: Is it possible to include the sst id and block id as well?

**Review by hzxa21 — APPROVED** [review]
Rest LGTM

**Review by zwang28 — APPROVED** [review]
LGTM

**Pull request merged**